### ISSUE TITLE
Improve layout

### ISSUE THREAD

**Issue by doup** [bevyengine/bevy-website#307]
Hi! I've seen the following layout related issues:

- https://github.com/bevyengine/bevy-website/issues/135
- https://github.com/bevyengine/bevy-website/issues/213
- https://github.com/bevyengine/bevy-website/issues/188

I think all of them can be improved at the same time if the layout is revisited. I've been looking around and I think [Docusaurus](https://docusaurus.io/docs) solution is quite good. It would allow:

- Header elements in desktop that hide into on the hamburger menu on mobile
- Call to actions (e.g. donate, get started), which stay visible in the header on mobile (until a breakpoint)
- Documentation menu that plays nicely with the main menu

Here is what I'm referring to:

https://user-images.githubusercontent.com/188612/160906129-5a71f62d-04ca-4070-aa31-ec68423473b5.mp4

- This is how the menu looks like on the `docs` section
- It clashes with the main menu, so it has two "levels" on the menu: docs menu & main menu
  - This could be applied only for the docs section
- When clicking on the hamburger the documentation menu shows
- But, the main menu can also be accessed with the `<-- Back to main menu` option

I could give it a try. Any thoughts?

**Comment by alice-i-cecile** [comment]
I like that style, and I think this is a good design to learn from :) I agree that these should be tackled at once.

**Comment by cart** [comment]
> It clashes with the main menu, so it has two "levels" on the menu: docs menu & main menu 

"Multiple hierarchies of menus" does bury top level navigation info in a way im not ecstatic about, but we do need to make a compromise somewhere and this is the best I've seen so far. I also like that this buys us space for doc-specific functionality like search buttons (and space for the bevy logo).

I think if we make tapping the bevy logo link to the hoe page,  that might help improve the discoverability of the "buried" content.

In short: im on board.

**Comment by IceSentry** [comment]
Are you saying that you want to integrate Docusaurus or that you want to implement a solution that looks like it?

**Comment by cart** [comment]
I read this as "implementing a solution that looks like it" / thats what I was agreeing to.

**Comment by doup** [comment]
Yeah, sorry if I wasn't clear, I meant I want to copy their UX solution for the doble-menu problem.

**Comment by doup** [comment]
Hi!

**I've a working prototype**, and as I was requested on Discord: no JavaScript, just SCSS+HTML. Just be aware that the styling isn't done (e.g. the docs menu is a placeholder). The prototype is on a test-project so I'll need to migrate it to `bevy-website` repo. The recorded video is running on Firefox responsive tool. Also, I still need to validate it on a real phone/tabe & all the browsers. But I'm quite confident it'll work properly.

Btw, below I've attached a diff between the menu with & without the documentation part. The downside of this is that it will duplicate the documentation menu between mobile & desktop; but anyway, probably not a big issue.

https://user-images.githubusercontent.com/188612/161129039-f5d96caf-d299-45af-a9eb-182fdf5fb0be.mp4

<img width="2041" alt="image" src="https://user-images.githubusercontent.com/188612/161129076-65bf8ade-7c5a-4c65-9c72-aae925ef637f.png">

**Comment by IceSentry** [comment]
Can you show what it looks like when it's on full desktop? While I appreciate mobile support _a lot_, my guess is that most people will be browsing on a desktop

Also, the added shadow when the menu is opened seemed pretty dark. It might just be because there's no content behind so it's hard to see.

Other than that. I really like the direction this is going.

**Comment by IceSentry** [comment]
Wait, I now realise it's just the header part, I thought it also contained the book chapters.

**Comment by doup** [comment]
Yes, my idea is to work on the "shell" (header, desktop-menu, mobile-menu & maybe footer) and from there start to improve the docs & content layout. Probably calling the issue "layout improvements" is a stretch, maybe I should change the scope of this… 😅

**Comment by doup** [comment]
I've reduced the scope to header responsive-design improvement and to improve the interaction between the main menu & the docs/assets menu. Title & description tweaked.
